src/main/java/com/google/api/codegen/transformer/SampleTransformer.java

```diff
@@ -183,44 +188,41 @@ public void generateSamples(
+    // to generate the samples in various scenarios. Once all the configs have been migrated to
+    // use the SampleSpec, we can delete the code below as well as sample_code_init_fields.
+    String defaultId =
+        (sampleType() == SampleType.IN_CODE) ? "sample_code_init_field" : INIT_CODE_SHIM;
 
-      List<ValueSetAndTags> matchingValueSets =
-          methodConfig.getSampleSpec().getMatchingValueSets(form, sampleType);
-
-      // For backwards compatibility in the configs, we need to use sample_code_init_fields instead
-      // to generate the samples in various scenarios. Once all the configs have been migrated to
-      // use the SampleSpec, we can delete the code below as well as sample_code_init_fields.
-      String id = null;
-      if (sampleType == SampleType.IN_CODE) {
-        // for IN_CODE, have the source of truth be sample_code_init_fields for
-        // now even if otherwise configured
-        id = "sample_code_init_field";
-      } else if (!methodConfig.getSampleSpec().isConfigured()
-          // if not configured, make the sample_code_init_fields available to
-          // all sample types
-          || matchingValueSets.isEmpty()) { // ApiMethodView.initCode still needs to be set for now
-        id = INIT_CODE_SHIM;
-      }
-      if (id != null) {
-        SampleValueSet valueSet =
-            SampleValueSet.newBuilder()
-                .setParameters(
-                    SampleParameters.newBuilder()
-                        .addAllDefaults(methodConfig.getSampleCodeInitFields())
+    ImmutableList<ValueSetAndTags> defaultValueSets =
+        ImmutableList.of(
+            ValueSetAndTags.newBuilder()
+                .values(
+                    SampleValueSet.newBuilder()
+                        .setParameters(
+                            SampleParameters.newBuilder()
+                                .addAllDefaults(methodConfig.getSampleCodeInitFields())
+                                .build())
+                        .setId(defaultId)
+                        .setDescription("value set imported from sample_code_init_fields")
+                        .setTitle("Sample Values")
                         .build())
-                .setId(id)
-                .setDescription("value set imported from sample_code_init_fields")
-                .setTitle("Sample Values")
-                .build();
-        matchingValueSets =
-            ImmutableList.of(ValueSetAndTags.newBuilder().values(valueSet).regionTag("").build());
+                .regionTag("")
+                .build());
+
+    for (CallingForm form : callingForms) {
+      List<ValueSetAndTags> matchingValueSets =
+          methodConfig.getSampleSpec().getMatchingValueSets(form, sampleType());
+
+      if (sampleType() == SampleType.IN_CODE
+          || !methodConfig.getSampleSpec().isConfigured()
+          || matchingValueSets.isEmpty()) {
+        matchingValueSets = defaultValueSets;
```

Comment:
Is the intention here to overwrite the contents of `matchingValueSets` when `sampleType() == SampleType.IN_CODE`?